### PR TITLE
fix: modify coupon & voucher tabs

### PR DESCRIPTION
**Pull request by HLTVProxy**
- add "used" status tab in coupon
- add "notYet" & "used" status in vouchar
- add CouponCollectionAdminPage loading view
- modify CouponAdminCard styles

### CODE REVIEW

File: src/components/voucher/VoucherCollectionTabs.tsx [hunk 1]
```diff
@@ -16,13 +16,24 @@ const VoucherCollectionTabs: React.VFC<{ vouchers: VoucherProps[] }> = ({ vouche
...
+      name: formatMessage(usersMessages.tab.notYet),
+      isDisplay: voucher =>
+        (voucher.startedAt && voucher.startedAt.getTime() > Date.now() && !voucher.status.used) || false,
     },
     {
       key: 'unavailable',
-      name: formatMessage(commonMessages.status.expired),
-      isDisplay: voucher => !voucher.available,
+      name: formatMessage(usersMessages.tab.expired),
+      isDisplay: voucher => (voucher.endedAt && voucher.endedAt.getTime() < Date.now()) || false,
```
Comment (Ponchimeow) [inline src/components/voucher/VoucherCollectionTabs.tsx:31]: NITS: equivalent to `voucher.status.outdated` ?

File: src/pages/member/CouponCollectionAdminPage.tsx [hunk 2]
```diff
@@ -37,40 +37,47 @@ const CouponCollectionAdminPage: React.VFC = () => {
       key: 'expired',
       tab: formatMessage(usersMessages.tab.expired),
       coupons: coupons.filter(
-        coupon =>
-          (coupon.couponCode.couponPlan.endedAt && coupon.couponCode.couponPlan.endedAt.getTime() < Date.now()) ||
-          coupon.status.used,
+        coupon => coupon.couponCode.couponPlan.endedAt && coupon.couponCode.couponPlan.endedAt.getTime() < Date.now(),
```
Comment (Ponchimeow) [inline src/pages/member/CouponCollectionAdminPage.tsx:40]: NITS: equivalent to `coupon.status.outdated` ?

File: src/components/voucher/VoucherCollectionTabs.tsx [hunk 3]
```diff
@@ -16,13 +16,24 @@ const VoucherCollectionTabs: React.VFC<{ vouchers: VoucherProps[] }> = ({ vouche
...
+      isDisplay: voucher => voucher.available,
+    },
+    {
+      key: 'notYet',
+      name: formatMessage(usersMessages.tab.notYet),
+      isDisplay: voucher =>
+        (voucher.startedAt && voucher.startedAt.getTime() > Date.now() && !voucher.status.used) || false,
     },
     {
       key: 'unavailable',
```
Comment (Ponchimeow) [inline src/components/voucher/VoucherCollectionTabs.tsx:29]: NITS: could be expired, in conformity wit coupon

**Review by Ponchimeow — APPROVED** [review]

File: src/pages/member/CouponCollectionAdminPage.tsx (hunk 2 above)
Comment (HLTVProxy) [reply src/pages/member/CouponCollectionAdminPage.tsx:40]: not equl

sql from voucher_status

```
SELECT
  DISTINCT voucher.id AS voucher_id,
  (
    (
      (voucher_plan.started_at IS NOT NULL)
      AND (voucher_plan.started_at > now())
    )
    OR (
      (voucher_plan.ended_at IS NOT NULL)
      AND (voucher_plan.ended_at < now())
    )
  ) AS outdated,
  (t.order_id IS NOT NULL) AS used
```

File: src/components/voucher/VoucherCollectionTabs.tsx (hunk 1 above)
Comment (HLTVProxy) [reply src/components/voucher/VoucherCollectionTabs.tsx:31]: not equl

sql from voucher_status

```
SELECT
  DISTINCT voucher.id AS voucher_id,
  (
    (
      (voucher_plan.started_at IS NOT NULL)
      AND (voucher_plan.started_at > now())
    )
    OR (
      (voucher_plan.ended_at IS NOT NULL)
      AND (voucher_plan.ended_at < now())
    )
  ) AS outdated,
  (t.order_id IS NOT NULL) AS used
```

File: src/components/voucher/VoucherCollectionTabs.tsx (hunk 3 above)
Comment (HLTVProxy) [reply src/components/voucher/VoucherCollectionTabs.tsx:29]: Actually this code copy from coupon XD

**Pull request merged**